platform/linuxbsd/x11/display_server_x11.cpp

```diff
@@ -3562,8 +3562,7 @@ void DisplayServerX11::_handle_key_event(WindowID p_window, XKeyEvent *p_event,
 	// XLookupString returns keysyms usable as nice keycodes.
 	char str[256] = {};
 	XKeyEvent xkeyevent_no_mod = *xkeyevent;
-	xkeyevent_no_mod.state &= ~ShiftMask;
-	xkeyevent_no_mod.state &= ~ControlMask;
+	xkeyevent_no_mod.state = 0;
```

Comment:
This change completely breaks some layouts. Try DVORAK, it will return QWERTY key codes and Russian labels (picking it from a previous layout I guess, since DVORAK has bit 13 set in the state, which is group id).

For the reference: keyboard layouts config used to test:

![Screenshot From 2025-02-02 13-21-30](https://github.com/user-attachments/assets/705d1d00-1a22-43ea-8af8-bb43f2cccb2e)

> A State Field from an XKB State
> 
> Many events report the keyboard state in a single state field. Using XKB, a state field combines modifiers, group and the pointer button state into a single sixteen bit value as follows:
> 
>    Bits 0 through 7 (the least significant eight bits) of the effective state comprise a mask of type KEYMASK which reports the state modifiers.
>
>    Bits 8 through 12 comprise a mask of type BUTMASK which reports pointer button state.
>
>    Bits 13 and 14 are interpreted as a two-bit unsigned numeric value and report the state keyboard group.
>
>    Bit 15 (the most significant bit) is reserved and must be zero.

I guess we only want to unset modifier bits, not all of it, like:

```suggestion
	xkeyevent_no_mod.state &= 0xFF00;
```

Reply:
> You confirm that this change fixes other layouts, right?

With `xkeyevent_no_mod.state &= 0xFF00;` it seems to work fine in all cases I ahve tested.

Reply:
Unfortunately it breaks the NEO layout layer4 "mod4" or "m4" here: 
![image](https://github.com/user-attachments/assets/18dd4251-45c8-4697-bcbb-fe38414a53b6)

See https://www.neo-layout.org.

I noticed it no longer working with the new 4.4 release and `git bisect`ed it down to commit 140a63be25cbcf0abdc363be3a3e52ea2bf05e9c of this PR...